### PR TITLE
docs(behavior_path_planner): update goal planner docs

### PR DESCRIPTION
**Pull request by kosuke55**
## Description

update goal planner docs

image

### <samp>🤖 Generated by Copilot at 292c7b8</samp>

This pull request improves the documentation of the `goal planner` module in the `behavior path planner` package. It explains the roles and interactions of the different submodules and provides visual and textual examples. It also defines the conditions for starting and finishing the goal planning process.
## Interface changes

not applicable

## Effects on system behavior

not applicable

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/docs/behavior_path_planner_goal_planner_design.md [hunk 1]
```diff
@@ -67,18 +72,55 @@ RRTStar --o FreeSpacePullOver
...
 
 PullOverPath --o PullOverPlannerBase
 GoalCandidates --o GoalSeacherBase
 
 @enduml
 ```
 
+## start condition
+
+Each is activated when all conditions are met.
```
Comment (shmpwk) [inline planning/behavior_path_planner/docs/behavior_path_planner_goal_planner_design.md:85]:
```suggestion
Either one is activated when all conditions are met.
```
??

File: planning/behavior_path_planner/docs/behavior_path_planner_goal_planner_design.md [hunk 2]
```diff
@@ -67,18 +72,55 @@ RRTStar --o FreeSpacePullOver
...
+<img src="https://user-images.githubusercontent.com/39142679/237929950-989ca6c3-d48c-4bb5-81e5-e8d6a38911aa.png" width="600">
+
+#### pull over on shoulder lane
+
+- The distance between the goal and ego-vehicle is shorter than `minimum_request_length`.
+- Goal is set in the `road_shoulder`
+
+<img src="https://user-images.githubusercontent.com/39142679/237929941-2ce26ea5-c84d-4d17-8cdc-103f5246db90.png" width="600">
+
+### finish condition
```
Comment (shmpwk) [inline planning/behavior_path_planner/docs/behavior_path_planner_goal_planner_design.md:113]:
```suggestion
## finish condition
```
If aligned with "## start condition"

File: planning/behavior_path_planner/docs/behavior_path_planner_goal_planner_design.md [hunk 3]
```diff
@@ -67,18 +72,55 @@ RRTStar --o FreeSpacePullOver
...
 ```
 
+## start condition
+
+Each is activated when all conditions are met.
+
+### fixed_goal_planner
+
+- The distance between the goal and ego-vehicle is shorter than `minimum_request_length`.
+- route is set with `allow_goal_modification=false` which is the default value.
```
Comment (shmpwk) [inline planning/behavior_path_planner/docs/behavior_path_planner_goal_planner_design.md:90]:
```suggestion
- Route is set with `allow_goal_modification=false` by default.
```

File: planning/behavior_path_planner/docs/behavior_path_planner_goal_planner_design.md [hunk 4]
```diff
@@ -67,18 +72,55 @@ RRTStar --o FreeSpacePullOver
...
+
+<img src="https://user-images.githubusercontent.com/39142679/237929955-c0adf01b-9e3c-45e3-848d-98cf11e52b65.png" width="600">
+
+### rough_goal_planner
+
+#### pull over on road lane
+
+- The distance between the goal and ego-vehicle is shorter than `minimum_request_length`.
+- route is set with `allow_goal_modification=true`
```
Comment (shmpwk) [inline planning/behavior_path_planner/docs/behavior_path_planner_goal_planner_design.md:99]:
```suggestion
- Route is set with `allow_goal_modification=true`
```

File: planning/behavior_path_planner/docs/behavior_path_planner_goal_planner_design.md [hunk 5]
```diff
@@ -67,18 +72,55 @@ RRTStar --o FreeSpacePullOver
...
+  - we can set this option with [SetRoute](https://github.com/autowarefoundation/autoware_adapi_msgs/blob/main/autoware_adapi_v1_msgs/routing/srv/SetRoute.srv#L2) api service.
```
Comment (shmpwk) [inline planning/behavior_path_planner/docs/behavior_path_planner_goal_planner_design.md:100]:
```suggestion
  - We can set this option with [SetRoute](https://github.com/autowarefoundation/autoware_adapi_msgs/blob/main/autoware_adapi_v1_msgs/routing/srv/SetRoute.srv#L2) api service.
```

File: planning/behavior_path_planner/docs/behavior_path_planner_goal_planner_design.md [hunk 6]
```diff
@@ -67,18 +72,55 @@ RRTStar --o FreeSpacePullOver
...
+  - we support `2D Rough Goal Pose` with the key bind `r` in rviz, but in the future there will be a panel of tools to manipulate various Route API from rviz.
```
Comment (shmpwk) [inline planning/behavior_path_planner/docs/behavior_path_planner_goal_planner_design.md:101]:
```suggestion
  - We support `2D Rough Goal Pose` with the key bind `r` in RViz, but in the future there will be a panel of tools to manipulate various Route API from RViz.
```

File: planning/behavior_path_planner/docs/behavior_path_planner_goal_planner_design.md [hunk 7]
```diff
@@ -67,18 +72,55 @@ RRTStar --o FreeSpacePullOver
...
+- ego-vehicle is in the same lane as the goal.
+
+<img src="https://user-images.githubusercontent.com/39142679/237929950-989ca6c3-d48c-4bb5-81e5-e8d6a38911aa.png" width="600">
+
+#### pull over on shoulder lane
+
+- The distance between the goal and ego-vehicle is shorter than `minimum_request_length`.
+- Goal is set in the `road_shoulder`
```
Comment (shmpwk) [inline planning/behavior_path_planner/docs/behavior_path_planner_goal_planner_design.md:109]:
```suggestion
- Goal is set in the `road_shoulder`.
```

File: planning/behavior_path_planner/docs/behavior_path_planner_goal_planner_design.md (hunk 1 above)
Comment (kosuke55) [reply planning/behavior_path_planner/docs/behavior_path_planner_goal_planner_design.md:85]: tnaks!!

File: planning/behavior_path_planner/docs/behavior_path_planner_goal_planner_design.md [hunk 8]
```diff
@@ -67,18 +72,55 @@ RRTStar --o FreeSpacePullOver
...
+- Route is set with `allow_goal_modification=false` by default.
+
+<img src="https://user-images.githubusercontent.com/39142679/237929955-c0adf01b-9e3c-45e3-848d-98cf11e52b65.png" width="600">
+
+### rough_goal_planner
+
+#### pull over on road lane
+
+- The distance between the goal and ego-vehicle is shorter than `minimum_request_length`.
+- Route is set with `allow_goal_modification=true`
```
Comment (shmpwk) [inline planning/behavior_path_planner/docs/behavior_path_planner_goal_planner_design.md:99]:
```suggestion
- Route is set with `allow_goal_modification=true` .
```

File: planning/behavior_path_planner/docs/behavior_path_planner_goal_planner_design.md [hunk 9]
```diff
@@ -67,18 +72,55 @@ RRTStar --o FreeSpacePullOver
...
+#### pull over on shoulder lane
+
+- The distance between the goal and ego-vehicle is shorter than `minimum_request_length`.
+- Goal is set in the `road_shoulder`.
+
+<img src="https://user-images.githubusercontent.com/39142679/237929941-2ce26ea5-c84d-4d17-8cdc-103f5246db90.png" width="600">
+
+## finish condition
+
+- The distance to the goal from your vehicle is lower than threshold (default: < `1m`)
```
Comment (shmpwk) [inline planning/behavior_path_planner/docs/behavior_path_planner_goal_planner_design.md:115]:
```suggestion
- The distance to the goal from your vehicle is lower than threshold (default: < `1m`).
```

**Review by shmpwk — APPROVED** [review]

**Review by satoshi-ota — APPROVED** [review]

**Pull request merged**